src/Umbraco.Web.UI.Client/src/common/directives/components/localization/localize.directive.js

```diff
@@ -30,16 +32,30 @@ angular.module("umbraco.directives")
     .directive('localize', function ($log, localizationService) {
         return {
             restrict: 'E',
-            scope:{
+            scope: {
                 key: '@'
             },
             replace: true,
 
             link: function (scope, element, attrs) {
                 var key = scope.key;
-                localizationService.localize(key).then(function(value){
-                    element.html(value);
-                });
+
+                if (!attrs["tokens"]) {
+                    localizationService.localize(key).then(function (value) {
+                        element.html(value);
+                    });
+                }
+                else {
+                    var unwatcher = scope.$watchCollection(function () { return scope.$parent.$eval(attrs["tokens"]) }, function (newValue, oldValue) {
```

Comment:
Should tokens just become part of the scope just like `key`? Generally accessing things like $parent anywhere should be avoided where possible. This would also mean the watch doesn't need to be a function and could just be "tokens"